docs/content/posts/2024-07-15-release-1.20.0.md

```diff
@@ -0,0 +1,545 @@
+
+Many SQL vendors expose the concepts of `Partitioning`, `Bucketing`, and `Clustering`. We propose to introduce
+the concept of `Bucketing` to Flink.
+
+Buckets enable load balancing in an external storage system by splitting data into disjoint subsets. It depends
+heavily on the semantics of the underlying connector. However, a user can influence the bucketing behavior by
+specifying the number of buckets, the bucketing algorithm, and (if the algorithm allows it) the columns which 
+are used for target bucket calculation. All bucketing components (i.e. bucket number, distribution algorithm, bucket key columns)
+are optional from a SQL syntax perspective.
+
+Take the following SQL statements as an example:
+
+```sql
+-- declares a hash function on a fixed number of 4 buckets (i.e. HASH(uid) % 4 = target bucket).
+CREATE TABLE MyTable (uid BIGINT, name STRING) DISTRIBUTED BY HASH(uid) INTO 4 BUCKETS;
+
+-- leaves the selection of an algorithm up to the connector.
+CREATE TABLE MyTable (uid BIGINT, name STRING) DISTRIBUTED BY (uid) INTO 4 BUCKETS;
+
+-- leaves the number of buckets up to the connector.
+CREATE TABLE MyTable (uid BIGINT, name STRING) DISTRIBUTED BY (uid);
+
+-- only defines the number of buckets.
+CREATE TABLE MyTable (uid BIGINT, name STRING) DISTRIBUTED INTO 4 BUCKETS;
+```
+
+**More Information**
+* [FLINK-33494](https://issues.apache.org/jira/browse/FLINK-33494)
+* [FLIP-376](https://cwiki.apache.org/confluence/display/FLINK/FLIP-376%3A+Add+DISTRIBUTED+BY+clause)
+
+# State & Checkpoint Improvements
+
+## Unified File Merging Mechanism for Checkpoints
+
+The unified file merging mechanism for checkpointing is introduced to Flink 1.20 as an MVP ("minimum viable product") feature,
+which allows scattered small checkpoint files to be written into larger files, reducing the number of file creations
+and file deletions and alleviating the pressure of file system metadata management raised by the file flooding problem
+during checkpoints.
+
+The mechanism can be enabled by setting `execution.checkpointing.file-merging.enabled` to `true`. For more advanced options
+and principle behind this feature, please refer to the Checkpointing documentation.
+
+**More Information**
+* [FLINK-33494](https://issues.apache.org/jira/browse/FLINK-32070)
+* [FLIP-306](https://cwiki.apache.org/confluence/display/FLINK/FLIP-306%3A+Unified+File+Merging+Mechanism+for+Checkpoints)
+* [Documentation](https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/dev/datastream/fault-tolerance/checkpointing/#unify-file-merging-mechanism-for-checkpoints-experimental)
+
+## Manually Compact Small SST Files
+
+In some cases, the number of files produced by RocksDB state backend grows indefinitely. This might cause task state
+info (TDD and checkpoint ACK) to exceed RPC message size and fail recovery/checkpoint in addition to having lots of small files.
+
+In Flink 1.20, you can manually merge such files in the background using RocksDB API.
```

Comment:
```suggestion
In some cases, the number of files produced by RocksDB state backend grows indefinitely.  In addition to the overhead of having lots of small files, this might cause the task state info to exceed RPC message size limit and result in recovery or checkpoint failures. 
Flink 1.20 can merge small files in the background using the RocksDB API.
```